fw/http_msg.c

```diff
@@ -1429,6 +1439,28 @@ __tfw_http_msg_expand_from_pool(TfwHttpTransIter *mit, TfwPool* pool,
 			 */
 			n_copy = room == 0 ? rlen : min(room, rlen);
 			off = c->len - rlen;
+			skb_room = SS_SKB_MAX_DATA_LEN - it->skb->len;
+			nr_frags = skb_shinfo(it->skb)->nr_frags;
+
+			if (skb_room == 0 || (it->skb == it->skb_head
+					      && it->frag == -1
+					      && nr_frags == MAX_SKB_FRAGS))
+			{
+				struct sk_buff *nskb;
+
+				nskb = ss_skb_alloc(0);
+				if (!nskb)
+					return -ENOMEM;
+				skb_shinfo(nskb)->tx_flags =
+					skb_shinfo(it->skb)->tx_flags;
+				skb_insert_after(it->skb, nskb);
+				skb_shinfo(nskb)->nr_frags = 0;
```

Comment:
We have the same code in `__extend_pgfrags()` - maybe move it to an inline function in `ss_skb.h`?